[Plugin] MoreSearchEngines

ImpishMoxxie opened the pull request:
## Allows you to search in a few different search engines!

**Only searches text from msgs**

plugin request thingy: https://github.com/Vencord/plugin-requests/issues/117

![img](https://i.imgur.com/qRxJTJT.png)

yes we added some funny sites :3

Vendicated commented on src/plugins/MoreSearchEngines/index.tsx:
```diff
@@ -0,0 +1,86 @@
+                                    aria-hidden="true"
+                                    height={16}
+                                    width={16}
+                                    src={new URL("/favicon.ico", Engines[engine]).toString().replace("lens.", "")}
```
```suggestion
                                    src={new URL("/favicon.ico", Engines[engine]).toString()}
```
none of your engines contain lens. (anymore?)

Vendicated commented on src/plugins/MoreSearchEngines/index.tsx:
```diff
@@ -0,0 +1,86 @@
+}
+
+const messageContextMenuPatch: NavContextMenuPatchCallback = (children, props) => () => {
+    const group = findGroupChildrenByChildId("copy-link", children);
```
```suggestion
	if (!message) return;

    const group = findGroupChildrenByChildId("copy-link", children);
```

Vendicated commented:
double search... why not replace the native discord option instead?

![image](https://github.com/Vendicated/Vencord/assets/45497981/9c8cba18-3251-4678-9e49-d305b33b6437)

Kubuxu commented:
I was looking at writing a plugin like this but for a slightly different use path.
I wanted to replace the `Search with Google` with my preferred search engine.

Also, could you add Kagi to the list? `https://kagi.com/search?q=`

ImpishMoxxie commented:
> I was looking at writing a plugin like this but for a slightly different use path. I wanted to replace the `Search with Google` with my preferred search engine.
> 
> Also, could you add Kagi to the list? `https://kagi.com/search?q=`


we made it so that now it's under the "search with Google" and we are working on making it only search the highlighted part.
also ya I can add kagi.com

ImpishMoxxie commented:
updated it to show under the "search with Google" as https://github.com/Vendicated/Vencord/pull/2218#issuecomment-1972188521 asked

added settings to select what engines to use

only shows now when marking text and uses the marked text for the search

Vendicated commented on src/plugins/MoreSearchEngines/index.tsx:
```diff
@@ -0,0 +1,137 @@
+const testieWestie: boolean = true;
+
+const Engines = {
+    DuckDuckGo: "https://duckduckgo.com/",
```
where's Google?

ImpishMoxxie replied:
google still shows this is just extra browsers

![image](https://i.imgur.com/8wN9DVP.png)

Vendicated commented:
> updated it to show under the "search with Google" as #2218 (comment) asked

that's absolutely not what i asked.. i suggested to **replace** the Google search option
